lib/phoenix_component.ex

```diff
@@ -2014,6 +2014,96 @@ defmodule Phoenix.Component do
   defp form_method(method) when method in ~w(get post), do: {method, nil}
   defp form_method(method) when is_binary(method), do: {"post", method}
 
+  @doc """
+  Renders nested form inputs for associations or embeds.
+
+  [INSERT LVATTRDOCS]
+
+  This function is built on top of `Phoenix.HTML.Form.inputs_for/3`.
+
+  For more information about  options and how to build inputs, see `Phoenix.HTML.Form`.
+
+  ## Examples
+
+  ```heex
+  <.form
+    :let={f}
+    for={@changeset}
+    phx-change="change_name"
+  >
+    <.inputs_for :let={f_nested} field={{:f, :nested}}>
+      <%= text_input f_nested, :name %>
+    </.inputs_for>
+  </.form>
+  ```
+  """
+  attr.(:field, :any,
+    required: true,
+    doc: "A %Phoenix.HTML.Form{}/field name tuple, for example: {f, :email}."
+  )
+
+  attr.(:id, :string,
+    doc: """
+    The id to be used in the form, defaults to the concatenation of the given
+    field to the parent form id.
+    """
+  )
+
+  attr.(:as, :atom,
+    doc: """
+    The name to be used in the form, defaults to the concatenation of the given
+    field to the parent form name.
+    """
+  )
+
+  attr.(:default, :any, doc: "The value to use if none is available.")
+
+  attr.(:prepend, :list,
+    doc: """
+    The values to prepend when rendering. This only applies if the field value
+    is a list and no parameters were sent through the form.
+    """
+  )
+
+  attr.(:append, :list,
+    doc: """
+    The values to append when rendering. This only applies if the field value
+    is a list and no parameters were sent through the form.
+    """
+  )
+
+  attr.(:skip_hidden, :boolean,
+    default: false,
+    doc: """
+    Skip the automatic rendering of hidden fields to allow for more tight control
+    over the generated markup. You can access `form.hidden` or use
+    `Phoenix.HTML.Form.hidden_inputs_for/1` to generate them manually.
+    """
+  )
+
+  slot.(:inner_block, required: true, doc: "The content rendered for each nested form.")
+
+  def inputs_for(assigns) do
+    {form, field} = assigns[:field] || raise ArgumentError, "missing :field assign to inputs_for"
+
+    opts =
+      assigns
+      |> Map.take([:id, :as, :default, :append, :prepend])
+      |> Enum.reject(fn {_, v} -> v == nil end)
+
+    assigns =
+      assigns
+      |> assign(:field, nil)
+      |> assign(:inputs, Phoenix.HTML.Form.inputs_for(form, field, opts))
+
+    ~H"""
+    <%= for finner <- @inputs do %>
```

Comment:
What is "finner"?

Reply:
It’s a variable for the potentially multiple inputs that will be rendered within inputs_for (like when used with an embeds_many).
The code has changed quite a bit since this pull request, where it seems to mean f(orms_)inner, see https://github.com/phoenixframework/phoenix_live_view/blob/3d46bdf6916119727897463413c6f11c06b15d63/lib/phoenix_component.ex#L2527-L2579

Reply:
Yeah that's it. `inputs_for` builds individual `Phoenix.HTML.Form` structs for each nested record, which has all the necessary field values namespaced correctly, e.g. names, ids, …. See `Phoenix.HTML.FormData.to_form/4`

Reply:
This code used to live in phoenix_html. I just moved it over to here. I don't think there's a reason to not change it to a more descriptive name.